libsolidity/codegen/ir/IRGenerator.cpp

```diff
@@ -583,7 +583,9 @@ string IRGenerator::dispatchRoutine(ContractDefinition const& _contract)
 	Whiskers t(R"X(
 		if iszero(lt(calldatasize(), 4))
 		{
-			let selector := <shr224>(calldataload(0))
+			mstore(0, 0)
```

Comment:
This is just a copy from #7597.

Reply:
I also added a comment to #7597: would it make sense to generate here different code in the evm/ewasm case?